File: Chapter05/FontSizes/FontSizes/FontSizes/FontSizesPage.cs
using System;
using Xamarin.Forms;

namespace FontSizes
{
    public class FontSizesPage : ContentPage
    {
        public FontSizesPage()
        {
            BackgroundColor = Color.White;
            StackLayout stackLayout = new StackLayout
            {
                HorizontalOptions = LayoutOptions.Center,
                VerticalOptions = LayoutOptions.Center
            };

            // Do the NamedSize values.
            NamedSize[] namedSizes = 
            {
                NamedSize.Default, NamedSize.Micro, NamedSize.Small,
                NamedSize.Medium, NamedSize.Large
            };

            foreach (NamedSize namedSize in namedSizes)
            {
                double fontSize = Device.GetNamedSize(namedSize, typeof(Label));

                stackLayout.Children.Add(new Label
                    {
                        Text = String.Format("Named Size = {0} ({1:F2})",
                                             namedSize, fontSize),
                        FontSize = fontSize,
                        TextColor = Color.Black
                    });
            }

            // Resolution in device-independent units per inch.
            double resolution = 160;

            // Draw horizontal separator line.
            stackLayout.Children.Add(
                new BoxView
                {
                    Color = Color.Accent,
                    HeightRequest = resolution / 80
                });

            // Do some numeric point sizes.
            int[] ptSizes = { 4, 6, 8, 10, 12 };

            foreach (double ptSize in ptSizes)
            {
                double fontSize = resolution * ptSize / 72;

                stackLayout.Children.Add(new Label
                    {
                        Text = String.Format("Point Size = {0} ({1:F2})",
                                             ptSize, fontSize),
                        FontSize = fontSize,
                        TextColor = Color.Black
                    });
            }

            Content = stackLayout;
        }
    }
}

File: Chapter05/FS/FontSizes/FontSizes/FontSizes/FontSizesPage.fs
namespace FontSizes

open System
open Xamarin.Forms

type FontSizesPage() = 
    inherit ContentPage(BackgroundColor = Color.White)

    let stackLayout = StackLayout(HorizontalOptions = LayoutOptions.Center,
                                  VerticalOptions = LayoutOptions.Center)

    do [ NamedSize.Default; NamedSize.Micro; NamedSize.Small; NamedSize.Medium; NamedSize.Large ]
        |> List.iter (fun namedSize -> let fontSize = Device.GetNamedSize(namedSize, typeof<Label>)
                                       stackLayout.Children.Add(Label(Text = String.Format("Named Size = {0} ({1:F2})",
                                                                                           namedSize, fontSize),
                                                                      FontSize = fontSize,
                                                                      TextColor = Color.Black)))
    let resolution = 160.0 

    do stackLayout.Children.Add(BoxView(Color = Color.Accent,
                                        HeightRequest = resolution / 80.0))


    do [ 4.0; 6.0; 8.0; 10.0; 12.0 ]
        |> List.iter (fun pointSize -> let fontSize = resolution * pointSize / 72.0
                                       stackLayout.Children.Add(Label(Text = String.Format("Point Size = {0} ({1:F2})",
                                                                                           pointSize, fontSize),
                                                                      FontSize = fontSize,
                                                                      TextColor = Color.Black)))
    do base.Content <- stackLayout

